compiler/plugins/target/ROCM/ROCMTarget.cpp

```diff
@@ -60,14 +60,12 @@ namespace mlir::iree_compiler::IREE::HAL {
 namespace {
 
 // TODO(#18792): rename flags back to iree-rocm- as they are not HIP-specific.
-// Only iree-hip-legacy-sync applies uniquely to HIP.
 struct ROCMOptions {
   std::string target = "";
   std::string targetFeatures = "";
   std::string bitcodeDirectory = getDefaultBitcodeDirectory();
   int wavesPerEu = 0;
   std::string enableROCMUkernels = "none";
-  bool legacySync = true;
```

Comment:
Digging through history:

* https://github.com/iree-org/iree/pull/16493 removed legacy_sync from all but ROCM, since "The ROCM HAL does not support command buffers at all"
* At the time, we still had hal/rocm and experimental/hip
* https://github.com/iree-org/iree/pull/15910 added graph command buffers to experimental/hip
* https://github.com/iree-org/iree/pull/16706 promoted experimental/hip to hal/drivers/
* We later deleted hal/rocm

@benvanik @nithinsubbiah does that sound right?

So I think we're ready for this now? All tests seem to be passing.